src/Cake.Common/Build/TeamCity/TeamCityDisposableHelpers.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cake.Common.Build.TeamCity
+{
+    /// <summary>
+    /// Disposable helper for writing TeamCity message blocks.
+    /// </summary>
+    public sealed class TeamCityBlock : IDisposable
```

Comment:
I really like this :smile:

1. Maybe the `StartBlock` method could be removed and done explicitly in the method that creates the block. This way we keep all the logic in one place.
2. Maybe we could make one generic block class that that takes an `Action<ITeamCityProvider>` that you pass to the constructor (for writing the end block).
3. Maybe an explicit implemented `IDisposable` interface that calls an EndBlock method or something so you also have the option to not use the using pattern.